Engine_Revit_UI/Modify/SetParameter.cs

```diff
@@ -133,9 +133,49 @@ public static bool SetParameter(this Element element, BuiltInParameter builtInPa
 
         public static bool SetParameter(this Parameter parameter, object value, Document document = null)
         {
+            // Skip null and read-only parameters
             if (parameter == null || parameter.IsReadOnly)
                 return false;
 
+            // Workset parameters
+            // Filter for worksets
+            if (parameter.Id.IntegerValue == (int)BuiltInParameter.ELEM_PARTITION_PARAM)
+            {
+                // Skip empty values
+                if (string.IsNullOrWhiteSpace(value as string))
+                    return false;
+
+                // Find all user worksets
+                FilteredWorksetCollector worksets =
+                    new FilteredWorksetCollector(document)
+                    .OfKind(WorksetKind.UserWorkset);
+
+                // Does a workspace with the specified name exist?
+                foreach (Workset workset in worksets)
```

Comment:
I would remove this and replace with something like the following - just less computationally expensive than a `for` loop going over all elements when one might not exist. Also less code to maintain/read :smile:

```
Workset workset = worksets.Where(x => x.Name == (value as string)).FirstOrDefault(); //Also need to check what happens if value is not a string

if(workset != null)
    return paremeter.Set(workset.Id.IntegerValue);
```

Reply:
Reimplemented in Linq as suggested